### PR TITLE
Add simple RBTree tests | RBTree size bugfix

### PR DESCRIPTION
**Pull request by admitrievtsev**
add: simple RBTree tests
- simple array test
- size of red black tree with duplicate keys
- simple array set test
- simple array set remove test
- adding and removing all elements test

fix: RBTree size decreasing after removing element fix

### CODE REVIEW

File: lib/src/main/kotlin/trees/implementations/RBTree.kt [hunk 1]
```diff
@@ -142,6 +142,7 @@ class RBTree<K : Comparable<K>, V> : BalanceBSTreeTemplate<K, V, RBVertex<K, V>>
                 vertex = vertex.right
             }
         }
+        size -= 1
```
Comment (odiumuniverse) [inline lib/src/main/kotlin/trees/implementations/RBTree.kt:145]: cringe

File: lib/src/test/kotlin/trees/rbtest/RBTest.kt [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        }
+
+        val expectedGet: Array<Int?> = emptyArray()
+        val actualGet: Array<Int?> = Array(rbTree.size, { i -> rbTree.get(i) })
+        assertContentEquals(expectedGet, actualGet, "Get method must return the value associated with the key")
+        val expectedSize = 0
+        val actualSize = rbTree.size
+        assertEquals(expectedSize, actualSize, "Size the tree must correspond to the number key-value pairs")
+    }
+}
```
Comment (odiumuniverse) [inline lib/src/test/kotlin/trees/rbtest/RBTest.kt]: criiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiinge

**Review by odiumuniverse — COMMENTED** [review]
)

File: lib/src/test/kotlin/trees/rbtest/RBTest.kt (hunk 2 above)
Comment (admitrievtsev) [reply lib/src/test/kotlin/trees/rbtest/RBTest.kt]: > criiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiinge

?

Comment (odiumuniverse) [reply lib/src/test/kotlin/trees/rbtest/RBTest.kt]: > criiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiinge

)

Comment (odiumuniverse) [reply lib/src/test/kotlin/trees/rbtest/RBTest.kt]: )

**Pull request merged**